NewSpring/corinth
# feat: Added Prayer Notifications

## DESCRIPTION

### What does this PR do, or why is it needed?
This PR adds the necessary capabilities to allow the app to directly open to one single prayer. This is so that we can display a prayer and it's answer when someone taps on a notification.

### How do I test this PR?
Run the app in the simulator.
Go to pray for a prayer and make sure that they look just like they always have.
Now close down the app. 

The best way I've found to test deep links like the one you need to test here is to set up a short cut in the Shortcuts app on the iOS simulator. 
![Screen Shot 2020-09-28 at 2 35 07 PM](https://user-images.githubusercontent.com/3229463/94472342-9ae09b80-0198-11eb-8e9e-258ef9b6ae80.png)
Open the Shortcuts app and add a new shortcut. Add the Open URL action and paste in `newspringchurchapp://AppStackNavigator/PrayerSingle?notification=true&id=780`. Give it a name and whatever prayer id you want it to have. Then tap this shortcut and it should open up the prayer in the app.

![Kapture 2020-09-28 at 14 46 55](https://user-images.githubusercontent.com/3229463/94472981-8fda3b00-0199-11eb-8add-5bb371efdc8a.gif)

As far as Android goes, I am having no luck with my emulator, whether I use `yarn android` from the command line or from Android Studio. My best suggestion here is to just go with testing iOS and then we'll test Android on an actual Android device once this gets merged. FWIW, when I tested the workflows and all that I got notifications on both my iPhone and Android device. So the biggest thing we'll need to test is whether or not the app opens correctly. 

## TODO


## REVIEW


**Manual QA**


---

> The purpose of PR Review is to _improve the quality of the software._
